.github/workflows/request_codeowners_review.yml

```diff
@@ -0,0 +1,44 @@
+---
+name: Request Code Owners Review
+
+on:
+  pull_request:
+    types: ["opened", "synchronize"]
+
+jobs:
+  auto_request_review:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: ${{ github.event.pull_request.commits + 1 }}
+
+      - name: Auto request review
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          reviewers=""
+          # Get changed files in this pull request
+          for changed_file in $(git diff --name-only ${{ github.event.before }} ${{ github.sha }}); do
+            echo "Checking $changed_file"
+            while read -r line; do
+              pattern=$(echo "$line" | awk '{print $1}')
+              owners=$(echo "$line" | awk '{$1=""; print $0}' | xargs)
+              # Remove leading / from pattern and changed_file for comparison
+              pattern="${pattern#/}"
+              changed_file="${changed_file#/}"
+              if [[ "$changed_file" == $pattern* ]]; then
+                reviewers="$reviewers $owners"
+              fi
+            done < .github/CODEOWNERS
+          done
+          reviewers=$(echo "$reviewers" | xargs -n1 | sort -u | xargs)
+          if [ -n "$reviewers" ]; then
+            reviewers_cleaned=${reviewers//@/}
+            reviewers_comma_separated=$(echo "$reviewers_cleaned" | tr ' ' ',')
+            echo "Requesting review from: $reviewers_comma_separated"
+            #echo gh pr edit ${{ github.event.pull_request.number }} --add-reviewer $reviewers_comma_separated
```

Comment:
commented code, is that intentional?

Reply:
Ooops, nope. That's how I tested it. Will fix.

Reply:
Fixed

Reply:
FWIW way I test GHA is i create fork and enable workflows there, for small actions it'll do :)